### ISSUE TITLE
"Unexpected data encountered in config data" when trying to create Braintree gateway

### ISSUE THREAD

**Issue by benface** [KuriousAgency/commerce-braintree#26]
When I try creating a gateway of type "Braintree" (regardless of the settings I enter), I'm getting the following error:

```
2020-03-24 23:58:46 [-][1][2epl5p5fb84p5cl7pgkh04g6ae][info][application] Unexpected data encountered in config data - Braintree\Gateway Object
(
    [config] => Braintree\Configuration Object
        (
            [_environment:Braintree\Configuration:private] => production
            [_merchantId:Braintree\Configuration:private] => 
            [_publicKey:Braintree\Configuration:private] => 
            [_privateKey:Braintree\Configuration:private] => 
            [_clientId:Braintree\Configuration:private] => 
            [_clientSecret:Braintree\Configuration:private] => 
            [_accessToken:Braintree\Configuration:private] => 
            [_proxyHost:Braintree\Configuration:private] => 
            [_proxyPort:Braintree\Configuration:private] => 
            [_proxyType:Braintree\Configuration:private] => 
            [_proxyUser:Braintree\Configuration:private] => 
            [_proxyPassword:Braintree\Configuration:private] => 
            [_timeout:Braintree\Configuration:private] => 60
            [_sslVersion:Braintree\Configuration:private] => 
            [_acceptGzipEncoding:Braintree\Configuration:private] => 1
        )

    [graphQLClient] => Braintree\GraphQLClient Object
        (
            [_service] => Braintree\GraphQL Object
                (
                    [_config:protected] => Braintree\Configuration Object
                        (
                            [_environment:Braintree\Configuration:private] => production
                            [_merchantId:Braintree\Configuration:private] => 
                            [_publicKey:Braintree\Configuration:private] => 
                            [_privateKey:Braintree\Configuration:private] => 
                            [_clientId:Braintree\Configuration:private] => 
                            [_clientSecret:Braintree\Configuration:private] => 
                            [_accessToken:Braintree\Configuration:private] => 
                            [_proxyHost:Braintree\Configuration:private] => 
                            [_proxyPort:Braintree\Configuration:private] => 
                            [_proxyType:Braintree\Configuration:private] => 
                            [_proxyUser:Braintree\Configuration:private] => 
                            [_proxyPassword:Braintree\Configuration:private] => 
                            [_timeout:Braintree\Configuration:private] => 60
                            [_sslVersion:Braintree\Configuration:private] => 
                            [_acceptGzipEncoding:Braintree\Configuration:private] => 1
                        )

                    [_useClientCredentials:Braintree\Http:private] => 
                )

        )

)
```

**Versions**
Craft CMS: 3.4.10.1
Craft Commerce: 3.0.12
Braintree for Craft Commerce: 2.3.4

**Comment by mcjackson18** [comment]
@benface this is now fixed in 2.3.5.

**Comment by benface** [comment]
Thank you! 🙌
